### ISSUE TITLE
[VC] - Webhook cabundle is using cluster's Root CA in stead of signing CSRs CA

### ISSUE THREAD

**Issue by jianh619** [kubernetes-sigs/multi-tenancy#1506]
I'm trying to deploy virtual cluster in openshift  , got an error of "x509: certificate signed by unknown authority" , this happens while apiserver trying to communicate with virtual-cluster webhook .

Then after checking the code , find the certificates are properly signed by [applying CSR](https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/virtualcluster/pkg/webhook/virtualcluster/virtualcluster_webhook.go#L201) , but the CAFile is using [root CA](https://github.com/kubernetes-sigs/multi-tenancy/blob/master/incubator/virtualcluster/pkg/webhook/virtualcluster/virtualcluster_webhook.go#L68) (normally we can find it in the service account's token secret ) , usually these two CA are some  , however in some platforms (like openshift , PKS ) , they are different , we can't just arbitrarily assert there're the same .

Env:  
Openshift 4.6.28
Kubernetes Version: v1.19.0+d856161

**Comment by charleszheng44** [comment]
Hi @jianh619  thanks for filing the bug.

**Comment by charleszheng44** [comment]
@jianh619 How does OpenShift handle this problem for other webhooks? I think this may be a generic issue. A potential solution would be storing the Apiserver CA in a secret, which can be used by the vc-manager when creating the validating webhook configuration, but I am not sure if this will cause any security issue.

**Comment by charleszheng44** [comment]
@jianh619 I just checked the source code of the [`WebhookClientConfig`](https://github.com/kubernetes/api/blob/648b77825832f4e96433407e4b406a3bdbb988bd/admissionregistration/v1/types.go#L533), looks like if the `CABundle` is unset, the APIServer will use the system trusts root. So I think maybe we can just leave it blank. I will try it out and get back to you if it works as expected.

**Comment by yaoyao12138** [comment]
@charleszheng44 I'm confused that when I follow your code changed way to re-build the vc-manager image and use it in my side, the error of "x509: certificate signed by unknown authority" happen again..  could help look at it? Thanks.  
```
➜  tenant1 kubectl vc create -f virtualcluster_1_nodeport.yaml -o $KUBECONFIG

I0514 02:04:37.685577 3782947 request.go:621] Throttling request took 1.198745241s, request: GET:https://api.crawly.cp.fyre.ibm.com:6443/apis/storage.k8s.io/v1?timeout=32s
I0514 02:04:47.885527 3782947 request.go:621] Throttling request took 11.398473174s, request: GET:https://api.crawly.cp.fyre.ibm.com:6443/apis/operators.coreos.com/v1alpha1?timeout=32s
create virtual cluster: Internal error occurred: failed calling webhook "virtualcluster.validating.webhook": Post "https://virtualcluster-webhook-service.vc-manager.svc:9443/validate-tenancy-x-k8s-io-v1alpha1-virtualcluster?timeout=30s": x509: certificate signed by unknown authority
```
```
➜  tenant1 oc logs vc-manager-b5587cbd5-xqwnx
2021/05/14 09:04:06 http: TLS handshake error from 10.254.8.1:57982: remote error: tls: bad certificate
2021/05/14 09:04:06 http: TLS handshake error from 10.254.8.1:57984: remote error: tls: bad certificate
{"level":"error","ts":1620983046.9601014,"logger":"controller-runtime.controller","msg":"Reconciler error","controller":"virtualcluster-controller","name":"vc-sample-1","namespace":"default","error":"Internal error occurred: failed calling webhook \"virtualcluster.validating.webhook\": Post \"https://virtualcluster-webhook-service.vc-manager.svc:9443/validate-tenancy-x-k8s-io-v1alpha1-virtualcluster?timeout=30s\": x509: certificate signed by unknown authority"}
```
```
➜  tenant1 oc get ValidatingWebhookConfiguration virtualcluster-validating-webhook-configuration -oyaml
...
webhooks:
- admissionReviewVersions:
  - v1beta1
  clientConfig:
    service:
      name: virtualcluster-webhook-service
      namespace: vc-manager
      path: /validate-tenancy-x-k8s-io-v1alpha1-virtualcluster
      port: 9443
  failurePolicy: Fail
  matchPolicy: Exact
  name: virtualcluster.validating.webhook
...
```
```
➜  tenant1 oc get po vc-manager-b5587cbd5-xqwnx -oyaml
...
containers:
  - args:
    - --disable-stacktrace=true
    - --enable-webhook=true
    command:
    - manager
    image: quay.io/yaoyao/virtualcluster:manager-amd64-caBuncle-blank
    imagePullPolicy: Always
    name: vc-manager
...
```

**Comment by charleszheng44** [comment]
@yaoyao12138 Sorry for the confusion, I am not an expert on OpenShift and do not have an OpenShift cluster at hand for test. PR #1507 allows APIServer to try out all available root CAs to authenticate the request sent from the vc validation webhook. May I know how does OpenShift sign the CSR? Is the CA accessible by the APIServer?

**Comment by yaoyao12138** [comment]
@charleszheng44 I opened two defects for the Openshift plantform, you could trace by these.   
https://bugzilla.redhat.com/show_bug.cgi?id=1960936   
https://bugzilla.redhat.com/show_bug.cgi?id=1960940  
Of course, I would check the  virtualcluster_1_nodeport.yaml to make sure this later, thanks.  
cc @jianh619

**Comment by jianh619** [comment]
Yes , I tried to manually apply webhookconfiguration , caBundle omited ,  according to [OCP doc](https://docs.openshift.com/container-platform/4.6/rest_api/extension_apis/validatingwebhookconfiguration-admissionregistration-k8s-io-v1.html) , it should generate the system trust ca , but nothing there , we can trace the bug  .

Thanks @charleszheng44 's for your quick action .
